granularflow.go

```diff
@@ -374,7 +387,9 @@ func (c *baseConn) optimistic(
 		case message.CommandComplete:
 			decodeCommandCompleteMsg(r)
 		case message.CommandDataDescription:
-			descs, err = decodeCommandDataDescriptionMsg(r, q, c)
+			var headers msgHeaders
+			descs, headers, err = c.decodeCommandDataDescriptionMsg(r, q)
+			c.cacheHeaders(q, headers)
```

Comment:
It is unclear to me from the [protocol documentation](https://www.edgedb.com/docs/internals/protocol/overview#granular-flow) if this is correct. The documentation says that if optimistic execute is run with erroneous IDs that the server will prepare the query and return a CommandDataDescription message. However, the CommandDataDescription message has no known headers listed. Will the headers generated from preparing the query be returned with the CommandDataDescription message in this case? The headers from PrepareComplete are important because they indicate weather a query is read only or not.

Reply:
It's a bug.  `CommandDataDescription` should be equivalent to `PrepareComplete`.  I'll make a fix.

Reply:
> `CommandDataDescription` should be equivalent to `PrepareComplete`

As far as the headers they may return?

Reply:
As far as the returned `CAPABILITY` header.